lib/events/filesessions/fileasync.go

```diff
@@ -274,8 +360,22 @@ func (u *Uploader) startUpload(fileName string) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	// Apparently, exclusive lock can be obtained only in RDWR mode on NFS
 	sessionFilePath := filepath.Join(u.cfg.ScanDir, fileName)
+	// Corrupted session records can clog the uploader
+	// that will indefinitely try to upload them.
+	isSessionError, err := u.checkSessionError(sessionID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if isSessionError {
+		return &sessionError{
+			err: trace.BadParameter(
+				"session recording %v is either corrupted or is using unsupported format, remove the file %v to correct the problem, remove the %v file to retry the upload",
+				sessionID, sessionFilePath, u.sessionErrorFilePath(sessionID)),
+		}
+	}
+
+	// Apparently, exclusive lock can be obtained only in RDWR mode on NFS
```

Comment:
How about having `func Scan` pre-filter all sessions that are marked as corrupted?
That way, you don't need the extra logic here and the handling of corrupt sessions is a little more consilidated

Reply:
Just tried to implement your suggestion and `startUpload` looks like a better place to deal with upload